limitador/src/limit.rs

```diff
@@ -34,12 +38,224 @@ pub struct Limit {
 
     // Need to sort to generate the same object when using the JSON as a key or
     // value in Redis.
-    #[serde(serialize_with = "ordered_set")]
-    conditions: HashSet<String>,
+    #[serde(serialize_with = "ordered_condition_set")]
+    conditions: HashSet<Condition>,
     #[serde(serialize_with = "ordered_set")]
     variables: HashSet<String>,
 }
 
+#[derive(Deserialize, Serialize, PartialEq, Eq, Debug, Clone, Hash)]
+#[serde(try_from = "String", into = "String")]
+pub struct Condition {
+    var_name: String,
+    predicate: Predicate,
+    operand: String,
+}
+
+#[derive(Debug)]
+pub struct ConditionParsingError {
+    error: SyntaxError,
+    pub tokens: Vec<Token>,
+    condition: String,
+}
+
+impl Display for ConditionParsingError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "Invalid condition: \"{}\"\n └ {}",
```

Comment:
Given how these are reported by serde:
```
Failed to load limit file: Invalid limit file: Couldn't parse: .[0].conditions: Invalid condition: "req.method == GET"
 └ SyntaxError: Unexpected token `Identifier: GET` at offset 15 at line 7 column 5
```

I'm thinking of suffixing it with ` for condition` so it'd read:
`Unexpected token `Identifier: GET` at offset 15 for condition at line 7 column 5` (as the `line` & `column` is where the condition field is, and `offset` is where the `SyntaxError` within that condition was found)

Reply:
Changed in [5f8a405](https://github.com/Kuadrant/limitador/pull/109/commits/5f8a405e83a52e1172d5ecd42d3115b106c48d32) to `Failed to load limit file: Invalid limit file: Couldn't parse: .[0].conditions: SyntaxError: Unexpected token `Identifier: GET` at offset 15 of condition "req.method == GET" at line 7 column 5`